Add support for react v18

```
npm ERR! Found: react@18.1.0
npm ERR! node_modules/react
npm ERR!   react@"^18.1.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^15.0.0 || ^16.0.0 || ^17.0.0" from react-latex-next@2.1.0
npm ERR! node_modules/react-latex-next
npm ERR!   react-latex-next@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

--- a/example/package.json
+++ b/example/package.json
@@ -5,8 +5,8 @@
   "private": true,
   "dependencies": {
     "prop-types": "^15.6.2",
-    "react": "^17.0.0",
-    "react-dom": "^17.0.0",
+    "react": "^18.0.0",
+    "react-dom": "^18.0.0",
     "react-scripts": "^1.1.4",
     "react-latex-next": "link:.."
   },

--- a/package.json
+++ b/package.json
@@ -27,7 +27,7 @@
     "prepare": "yarn run build"
   },
   "dependencies": {
-    "katex": "^0.13.0"
+    "katex": "^0.15.0"
   },
   "peerDependencies": {
     "prop-types": "^15.5.4",

--- a/package.json
+++ b/package.json
@@ -27,7 +27,7 @@
     "prepare": "yarn run build"
   },
   "dependencies": {
-    "katex": "^0.15.0"
+    "katex": "^0.13.0"
   },
   "peerDependencies": {
     "prop-types": "^15.5.4",

--- a/package.json
+++ b/package.json
@@ -31,8 +31,8 @@
   },
   "peerDependencies": {
     "prop-types": "^15.5.4",
-    "react": "^15.0.0 || ^16.0.0 || ^17.0.0",
-    "react-dom": "^15.0.0 || ^16.0.0 || ^17.0.0"
+    "react": "^15.0.0 || ^16.0.0 || ^17.0.0 || ^18.0.0",
+    "react-dom": "^15.0.0 || ^16.0.0 || ^17.0.0 || ^18.0.0"
   },
   "devDependencies": {
     "@types/enzyme": "^3.1.13",
@@ -48,8 +48,8 @@
     "enzyme-adapter-react-16": "^1.1.1",
     "enzyme-to-json": "^3.3.4",
     "gh-pages": "^1.2.0",
-    "react": "^17.0.0",
-    "react-dom": "^17.0.0",
+    "react": "^18.0.0",
+    "react-dom": "^18.0.0",
     "react-scripts-ts": "^2.16.0",
     "react-test-renderer": "^16.4.2",
     "rollup": "^0.62.0",

